File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) 
    {
        map<char,int> a,b;

        int n= s.size();
        int m= t.size();

        for(int i=0; i<n; i++)
        {
            a[s[i]]++;
        } 

        for(int i=0; i<m; i++)
        {
            b[t[i]]++;
        }       

        return a==b;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        if prices.len() <= 1 {
            return 0;
        }

        let mut i = 0;
        let mut j = 1;
        let mut ans = 0;

        while j < prices.len() 
        {
            if prices[j] > prices[i] 
            {
                ans = ans.max(prices[j] - prices[i]);
                j += 1;
            } 
            
            else 
            {
                i = j;
                j += 1;
            }
        }

        ans
    }
}